File: FarNetCSharp/Tool1.cs
using System;
using FarNet;

namespace FarNetCSharp
{
	[System.Runtime.InteropServices.Guid("f584e496-b801-4df3-be4f-2848d3a6e77c")]
	[ModuleTool(Name = "Hello.C#", Options = ModuleToolOptions.AllMenus)]
	public class Tool1 : ModuleTool
	{
		public override void Invoke(object sender, ModuleToolEventArgs e)
		{
			Far.Api.Message("Hello from C# module!");
		}
	}
}

File: FarNetFSharp/Tool1.fs
namespace FarNetFSharp
open FarNet

[<ModuleTool(Name = "Hello from FarNetFSharp", Options = ModuleToolOptions.AllMenus)>]
[<Guid("cc0ecb2f-552a-4fe4-90f0-227077386890")>]
type Tool1() =
    inherit ModuleTool()
    override _.Invoke(_, e) =
        Far.Api.Message $"Hello from {e.From}"
